File: benchmarks/pairs/vfc-js/portprocesses-1-0-4/fixed.js
// Provenance: rrainn/PortProcesses  (fixed).
// repo: rrainn/PortProcesses
// commit: 86811216c9b97b01b5722f879f8c88a7aa4214e1
// parent: fffceb09aff7180afbd0bd172e820404b33c8299
// commit_url: https://github.com/rrainn/PortProcesses/commit/86811216c9b97b01b5722f879f8c88a7aa4214e1
// cve: CVE-2021-23348
// license: MIT
// function: listProcessesOnPort
// relpath: index.js
// provenance: human
// mechanism: source_reaches_sink
// upstream_start: 2

const listProcessesOnPort = module.exports.listProcessesOnPort = async port => {
	const portNumber = parseInt(port, 10);
	if (Number.isNaN(portNumber)) {
		console.error("Must provide number for port.");
		return;
	}
	try {
		const result = (await exec(`lsof -i :${portNumber}`)).output.split('\n');
		const headers = result.shift().split(' ').filter(item => !!item.trim() && item.trim() !== "").map(item => item.toLowerCase());
		return result.filter(item => !!item.trim() && item.trim() !== "").reduce((accumulator, currentValue) => {
			accumulator.push(currentValue.split(' ').filter(item => !!item.trim() && item.trim() !== "").reduce((accumulator, currentValue, index) => {
				if (index > headers.length - 1) {
					accumulator[headers[headers.length - 1]] = (!!accumulator[headers[headers.length - 1]].trim() && accumulator[headers[headers.length - 1]].trim() !== "") ? `${accumulator[headers[headers.length - 1]]} ${currentValue}` : currentValue;
				} else {
					accumulator[headers[index]] = currentValue;
				}
				return accumulator;
			}, {}));
			return accumulator;
		}, []);
	} catch (e) {
		console.error(e);
	}
};

File: benchmarks/pairs/vibe-py/vfapi-main-sql-return-users-from-username/fixed.py
# Provenance: naryal2580/vfapi sql_return_users_from_username (fixed).
# repo: naryal2580/vfapi
# commit: f36f177e1a32aa49272f4eb52f8be0af3e8f6bc3
# parent: f36f177e1a32aa49272f4eb52f8be0af3e8f6bc3
# commit_url: https://github.com/naryal2580/vfapi/blob/f36f177e1a32aa49272f4eb52f8be0af3e8f6bc3/main.py#L182
# cve: 
# license: MIT
# function: sql_return_users_from_username
# relpath: main.py
# provenance: human
# mechanism: source_reaches_sink

async def nosql_return_users_from_username(username: str):
    return get_nosql_users({'username': username})
